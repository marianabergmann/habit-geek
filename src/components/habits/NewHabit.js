import React, { useState } from "react";

const NewHabit = ({ createHabit }) => {
  const [name, setName] = useState("");
  const [type, setType] = useState("");
  const [frequency, setFrequency] = useState("");

  return (
    <div>
      <h2>Ready to start a new habit?</h2>
      <div>
        <label htmlFor="new-habit-name">
          Name
          <input
            id="new-habit-name"
            value={name}
            onChange={e => setName(e.target.value)}
          ></input>
        </label>
      </div>
      <div>
        <label htmlFor="new-habit-type">
          Type
          <select
            id="new-habit-type"
            value={type}
            onChange={e => setType(e.target.value)}
            onBlur={e => setType(e.target.value)}
          >
            <option vaue=""></option>
            <option vaue="Health">Health</option>
            <option vaue="Social">Social</option>
            <option vaue="Career">Career</option>
            <option vaue="Hobbies">Hobbies</option>
          </select>
        </label>
      </div>
      <div>
        <label htmlFor="new-habit-frequency">
          Frequency
          <select
            id="new-habit-frequency"
            value={frequency}
            onChange={e => setFrequency(e.target.value)}
            onBlur={e => setFrequency(e.target.value)}
          >
            <option vaue=""></option>
            <option vaue="1x">1x</option>
            <option vaue="2x">2x</option>
            <option vaue="3x">3x</option>
            <option vaue="4x">4x</option>
            <option vaue="5x">5x</option>
            <option vaue="6x">6x</option>
            <option vaue="7x">7x</option>
          </select>
        </label>
      </div>
      <button onClick={() => createHabit(name, type, frequency)}>
        Create!
      </button>
    </div>
  );
};

export default NewHabit;
