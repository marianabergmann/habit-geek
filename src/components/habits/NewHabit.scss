.new-habits {
  width: 100%;
  max-width: 340px;
  padding: 15px;
  margin: auto;
}

.new-habits__title {
  text-align: center;
}
