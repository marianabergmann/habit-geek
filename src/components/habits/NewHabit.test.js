import React from "react";
import { shallow } from "enzyme";
import NewHabit from "./NewHabit";

describe("NewHabits component", () => {
  it("Renders static elements", () => {
    const newHabits = shallow(<NewHabit></NewHabit>);
    expect(newHabits).toMatchSnapshot();
  });

  it("creates a habit", () => {
    const createHabitMock = jest.fn();
    const newHabits = shallow(
      <NewHabit createHabit={createHabitMock}></NewHabit>
    );
    newHabits
      .find("#new-habit-name")
      .simulate("change", { target: { value: "Read" } });
    newHabits
      .find("#new-habit-type")
      .simulate("change", { target: { value: "Social" } });
    newHabits
      .find("#new-habit-frequency")
      .simulate("change", { target: { value: "1x" } });
    newHabits.find("button").simulate("click");
    expect(createHabitMock).toHaveBeenCalledWith("Read", "Social", "1x");
  });

  it("selects change on blur for accessibility", () => {
    const createHabitMock = jest.fn();
    const newHabits = shallow(
      <NewHabit createHabit={createHabitMock}></NewHabit>
    );
    newHabits
      .find("#new-habit-type")
      .simulate("blur", { target: { value: "Social" } });
    newHabits
      .find("#new-habit-frequency")
      .simulate("blur", { target: { value: "1x" } });
    newHabits.find("button").simulate("click");
    expect(createHabitMock).toHaveBeenCalledWith("", "Social", "1x");
  });
});
