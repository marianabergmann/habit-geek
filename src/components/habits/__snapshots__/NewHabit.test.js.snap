// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewHabits component Renders static elements 1`] = `
<div
  className="new-habits"
>
  <h2
    className="new-habits__title"
  >
    Ready to start a new habit?
  </h2>
  <div
    className="mb-3"
  >
    <label
      htmlFor="new-habit-name"
    >
      Name
    </label>
    <input
      className="form-control"
      id="new-habit-name"
      onChange={[Function]}
      value=""
    />
  </div>
  <div
    className="mb-3"
  >
    <label
      htmlFor="new-habit-type"
    >
      Type
    </label>
    <select
      className="form-control"
      id="new-habit-type"
      onBlur={[Function]}
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      />
      <option
        value="Health"
      >
        Health
      </option>
      <option
        value="Social"
      >
        Social
      </option>
      <option
        value="Career"
      >
        Career
      </option>
      <option
        value="Hobbies"
      >
        Hobbies
      </option>
    </select>
  </div>
  <div
    className="mb-3"
  >
    <label
      htmlFor="new-habit-frequency"
    >
      Frequency
    </label>
    <select
      className="form-control"
      id="new-habit-frequency"
      onBlur={[Function]}
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      />
      <option
        value="1"
      >
        1x
      </option>
      <option
        value="2"
      >
        2x
      </option>
      <option
        value="3"
      >
        3x
      </option>
      <option
        value="4"
      >
        4x
      </option>
      <option
        value="5"
      >
        5x
      </option>
      <option
        value="6"
      >
        6x
      </option>
      <option
        value="7"
      >
        7x
      </option>
    </select>
  </div>
  <hr
    className="mb-4"
  />
  <button
    className="btn btn-primary btn-lg btn-block"
    onClick={[Function]}
  >
    Create!
  </button>
</div>
`;
